licensing/src/main/java/io/crate/license/LicenseKey.java

```diff
@@ -35,13 +35,44 @@
 import java.nio.ByteBuffer;
 import java.util.Base64;
 import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Objects;
 
 public class LicenseKey extends AbstractNamedDiffable<MetaData.Custom> implements MetaData.Custom {
 
     public static final String WRITEABLE_TYPE = "license";
 
-    static final int SELF_GENERATED = 0;
+    enum LicenseType {
+        SELF_GENERATED(0),
+        ENTERPRISE(1);
+
+        static Map<Integer, LicenseType> types = new HashMap<>();
+
+        static {
+            types.put(SELF_GENERATED.value, SELF_GENERATED);
+            types.put(ENTERPRISE.value, ENTERPRISE);
+        }
+
+        private int value;
+
+        LicenseType(int value) {
+            this.value = value;
+        }
+
+        static LicenseType of(Integer value) {
+            LicenseType type = types.get(value);
```

Comment:
I wonder if we should do a simple `switch/case` here instead of going through the map? Or simply use the Enum ordinals.

Reply:
Agree. Map is an overkill. Changed this to use a simple switch statement, I'm not that fond of using Enum ordinals, it can lead to issues (eg if the order changes in the future).